File: backend/app/schemas/knowledge.py
from typing import Any

from pydantic import BaseModel, Field

from app.knowledge.retriever import MAX_TOP_K


class KnowledgeSearchRequest(BaseModel):
    query: str = Field(..., min_length=1, max_length=2000)
    top_k: int = Field(default=5, ge=1, le=MAX_TOP_K)


class KnowledgeChunk(BaseModel):
    text: str
    score: float
    source: str
    article_title: str
    source_url: str
    document_id: str
    chunk_id: str
    chunk_index: int
    metadata: dict[str, Any]


class KnowledgeSearchResponse(BaseModel):
    query: str
    top_k: int
    embedding_model: str
    query_tokens: int
    latency_ms: int
    results: list[KnowledgeChunk]

File: frontend/src/types/knowledge.ts
export interface KnowledgeSearchRequest {
  query: string;
  top_k: number;
}

export interface KnowledgeChunk {
  text: string;
  score: number;
  source: string;
  article_title: string;
  source_url: string;
  document_id: string;
  chunk_id: string;
  chunk_index: number;
  metadata: Record<string, unknown>;
}

export interface KnowledgeSearchResponse {
  query: string;
  top_k: number;
  embedding_model: string;
  query_tokens: number;
  latency_ms: number;
  results: KnowledgeChunk[];
}
